packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx

```diff
@@ -38,26 +39,22 @@ export const useSignUpContext = (): SignUpContextType => {
   }
 
   const afterSignUpUrl = clerk.buildUrlWithAuth(
-    pickRedirectionProp('afterSignUpUrl', {
-      queryParams,
-      ctx,
-      options,
-      displayConfig,
-    }),
+    pickUrl(['afterSignUpUrl', 'redirectUrl'], queryParams, {
+      validator: url => isAllowedRedirectOrigin(url, options.allowedRedirectOrigins),
+      formatter: camelToSnake,
+    }) || pickUrl('afterSignUpUrl', [ctx, options, displayConfig]),
```

Comment:
❓ The priority for this before refactoring was:

1. queryParams.afterSignUpUrl
2. queryParams.redirectUrl
3. ctx.afterSignUpUrl
4. **ctx.redirectUrl**
5. options.afterSignUpUrl
6. displayConfig.afterSignUpUrl

Is the priority still respected? Especially `ctx.redirectUrl`.

Reply:
Yes it should. For every source, we check all the values, in that order.